Code-Standards-and-Ways-of-Working.md

```diff
@@ -0,0 +1,79 @@
+
+#### Running Test
+ChromaticQA runs as a part of the build. The builds can be seen here (https://www.chromaticqa.com/builds?appId=5d28eb3fe163f6002046d6fa)
+
+To run chromatic locally export the app code in bash profile(`export CHROMATIC_APP_CODE=<app-code>`) and run `npx chromatic test run  --build-script-name build:storybook || true`
+
+- When the tests are run first it creates the baselines screenshots unreviewed for all the browsers enabled in the settings.
+- Each new build creates snapshots of the components which are compared with the baseline screenshots and unchanged ones are automatically accepted and changed ones needs review(accordingly it has to be accepted or denied). If there are new components added they are listed under new components.
+- Each story has its own baseline that is tracked independently on each branch. When you approve a snapshot you also update the baseline for that story.
+
+Chromatic
+Advantages
+- We do not have to manage any infrastructure kind of stuff as it is maintained by the storybook people.
+- Run time for chrome and IE together is approx 6-8m.
+
+### Accessibility testing
+When you add a component to this repository, you should carry out an Accessibility Swarm, ideally including team members from multiple disciplines and using [the Assistive Technology we support](./README.md#assistive-technology-support).
+
+For changes to an existing component, a shorter swarm with fewer people should be conducted.
+
+[Here are detailed steps on how to run accessibility swarms.](https://bbc.github.io/accessibility-news-and-you/accessibility-swarms)
+
+As a result of the swarm, you may find some issues with your implementation. These should be documented and where relevant, [GitHub Issues](https://github.com/bbc/psammead/issues/new?template=bug_report.md) should be raised.
+
+### Storybook
+If the component is publicly available in Storybook you can preview this on any device. Don't forget to follow the [testing steps](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-assistive-technology) and use [supported assistive technology](https://bbc.github.io/accessibility-news-and-you/accessibility-and-supported-assistive-technology).
+
+#### Open the component in a tab
+Storybook publishes components in an iframe within the Storybook UI, this will be acknowledged by assistive technology such as a screen reader, it's therefore recommeded to **view/open the component in another tab** to the Storybook UI, you can then see the component code in isolation without the iframe and Storybook UI.
+
+### Local code in Storybook
+If you have a PR that is not yet merged and you need to preview this on another device, you can do so in Storybook via tunnelling:
+
+* Install packages ‘npm run install:packages’
+* Run storybook ‘npm run storybook’
+* Run your tunnelling tool on the Storybook port
```

Comment:
Should we give examples of tunnelling tools like `ngrok`?

Reply:
>Should we give examples of tunnelling tools like ngrok?

Definite no, that's a security hole in protected networks, stay away from that area.